### PR TITLE
ARROW-4410: [C++] Fix edge cases in InvertKernel

### PR DESCRIPTION
**Pull request by emkornfield**
(this also includes documentation fixes which should fix the build due to doxygen warnings) but they are duplciated from ARROW-4411 https://github.com/apache/arrow/pull/3518

### CODE REVIEW

File: cpp/src/arrow/compute/kernel.h [hunk 1]
```diff
@@ -162,6 +162,8 @@ class ARROW_EXPORT UnaryKernel : public OpKernel {
  public:
   /// \brief Executes the kernel.
   ///
+  /// \param[in,out] ctx The FunctionContext to use in th kernel.
```
Comment (pcmoritz) [inline cpp/src/arrow/compute/kernel.h]: th -> the

**Review by wesm — APPROVED** [review]
+1. Merge on rebase

**Comment by emkornfield** [comment]
CI failure looks like something related to tensor flow and pandas setup:

```
+python -c 'import pyarrow; import tensorflow'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/travis/build/apache/arrow/pyarrow-test-3.6/lib/python3.6/site-packages/tensorflow/__init__.py", line 24, in <module>
    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
  File "/home/travis/build/apache/arrow/pyarrow-test-3.6/lib/python3.6/site-packages/tensorflow/python/__init__.py", line 88, in <module>
    from tensorflow.python import keras
  File "/home/travis/build/apache/arrow/pyarrow-test-3.6/lib/python3.6/site-packages/tensorflow/python/keras/__init__.py", line 29, in <module>
    from tensorflow.python.keras import datasets
  File "/home/travis/build/apache/arrow/pyarrow-test-3.6/lib/python3.6/site-packages/tensorflow/python/keras/datasets/__init__.py", line 25, in <module>
    from tensorflow.python.keras.datasets import imdb
  File "/home/travis/build/apache/arrow/pyarrow-test-3.6/lib/python3.6/site-packages/tensorflow/python/keras/datasets/imdb.py", line 25, in <module>
    from tensorflow.python.keras.preprocessing.sequence import _remove_long_seq
  File "/home/travis/build/apache/arrow/pyarrow-test-3.6/lib/python3.6/site-packages/tensorflow/python/keras/preprocessing/__init__.py", line 30, in <module>
    from tensorflow.python.keras.preprocessing import image
  File "/home/travis/build/apache/arrow/pyarrow-test-3.6/lib/python3.6/site-packages/tensorflow/python/keras/preprocessing/image.py", line 23, in <module>
    from keras_preprocessing import image
  File "/home/travis/build/apache/arrow/pyarrow-test-3.6/lib/python3.6/site-packages/keras_preprocessing/image/__init__.py", line 8, in <module>
    from .dataframe_iterator import DataFrameIterator
  File "/home/travis/build/apache/arrow/pyarrow-test-3.6/lib/python3.6/site-packages/keras_preprocessing/image/dataframe_iterator.py", line 11, in <module>
    from pandas.api.types import is_numeric_dtype
ModuleNotFoundError: No module named 'pandas'
```

**Comment by emkornfield** [comment]
I don't know if the "merge on rebase" was supposed to do something automatically, that I messed up by doing a force push after rebasing directly against upstream/master (if it was a I apologize)

**Comment by emkornfield** [comment]
@wesm rebased.

**Pull request closed without merging**